src/Internal/Payment/PaymentRequest.php

```diff
@@ -144,4 +144,14 @@ public function get_payment_method(): PaymentMethodInterface {
 
 		throw new PaymentRequestException( __( 'No valid payment method was selected.', 'woocommerce-payments' ) );
 	}
+
+	/**
+	 * Populates a payment context before processing a payment.
```

Comment:
Maybe we can explain what needs to be populated and why so anyone who sees this function and comments are familiar with the whole flow.